contracts/Core/StakeManager.sol

```diff
@@ -183,13 +183,16 @@ contract StakeManager is Initializable, StakeStorage, StateManager, Pause, Stake
         require(locks[msg.sender][staker.tokenAddress].amount == 0, "Existing Lock");
         require(sAmount > 0, "Non-Positive Amount");
 
-        // slither-disable-next-line reentrancy-events,reentrancy-no-eth
-        rewardManager.giveInactivityPenalties(epoch, stakerId);
-
         IStakedToken sToken = IStakedToken(staker.tokenAddress);
         require(sToken.balanceOf(msg.sender) >= sAmount, "Invalid Amount");
+        uint256 totalSupply = sToken.totalSupply();
+
+        if ((staker.stake - _convertSRZRToRZR(sAmount, staker.stake, totalSupply)) < minStake) {
```

Comment:
`_convertSRZRToRZR(sAmount, staker.stake, totalSupply)` can instead be just `rAmount`? Code repitition

Reply:
hmm rAmount before giving penalties and after would change 
as stakers stake change
that's why we have to calculate twice I think

but it can be optimised 
have rAmount at top 
and only update when we penalise